src/js/components/githubIssues/resyncIssuesButton.tsx

```diff
@@ -0,0 +1,55 @@
+import Button from '@salesforce/design-system-react/components/button';
+import { t } from 'i18next';
+import React, { useCallback, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+
+import { LabelWithSpinner } from '@/js/components/utils';
+import { ThunkDispatch } from '@/js/store';
+import { refreshGitHubIssues } from '@/js/store/projects/actions';
+
+interface Props {
+  isRefreshing: boolean;
+  projectId: string;
+  forceResync: boolean;
+}
+
+const ResyncIssuesButton = ({
+  isRefreshing,
+  projectId,
+  forceResync,
+}: Props) => {
+  const dispatch = useDispatch<ThunkDispatch>();
+  const refreshIssues = useCallback(() => {
+    dispatch(refreshGitHubIssues(projectId));
+  }, [dispatch, projectId]);
+
+  // If there are no issues, resync with GitHub...
+  useEffect(() => {
+    if (forceResync) {
+      refreshIssues();
+    }
+  }, [forceResync, refreshIssues]);
```

Comment:
@oluoluoxenfree I added this to resync initially if we've fetched the issues from the API and there are none.